### DISCUSSION TITLE
Missing images?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5652] [Q&A]
Having issues getting images to show up in the frontend on a staging website, anyone come across this?

- Images work locally
- Images uploaded on staging site aren't showing up on the frontend?
- I can upload them via the CP, images presets are generated (checked via sFTP)
- I've checked the HTML source, copied the filename and checked it exists on the server - it does

For some reason the website isn't seeing the image file, even though the `src` is pointing to the file in the correct location...

I'm still digging into this, but thought I'd post this here in case anyone else has experienced this.

- Checked image (uploaded by client) isn't CMYK
- Tried to upload an image that works locally (edited filename) to the staging site

```
Statamic 3.3.2 Pro
Laravel 9.5.1
PHP 8.0.14
```

**Comment by duncanmcclean** [comment]
If you view source, then click into one of the images (there should be a link), does that work?

**Reply by stephenmeehanuk** [reply]
Thanks for the reply. No, it 404s.

**Reply by duncanmcclean** [reply]
What do your image URLs look like?

**Reply by stephenmeehanuk** [reply]
This one works
```
/img/containers/assets/projects/south-heywood/20019-south-heywood-commercial-cgi-1.jpg/557c6ff37c594070021379e1927c7bb7.jpg
```

This one doesn't
```
/img/containers/assets/projects/south-heywood/chester-wade-msnyz9L6gs4-unsplash.jpg/a1f99f969f8d6847cf3bdad9a7cf266e.jpg
```

**Reply by stephenmeehanuk** [reply]
Both files exist on the server, checked sFTP.

**Reply by duncanmcclean** [reply]
And if you try an image without loading it with Glide, my guess is that it works?

**Reply by stephenmeehanuk** [reply]
Yes! A basic `{{url}}` without glide preset works... But the presets are generated, they're on the server?

**Comment by jasonvarga** [comment]
Are they 404ing or is it a different error code?

**Reply by stephenmeehanuk** [reply]
Hi, they're 404ing when using a glide preset, but work when I use `url`

**Comment by stephenmeehanuk** [comment]
It's odd that `url` works but `glide:url` doesn't? The presets are generated and the `src` points to the correct directory and file on the server. The same is happening on another website (in development) on a different staging server. Both sites are running `3.3.2`. Both upgraded from `3.2.X`

A third site running `3.2.37` doesn't have this issue. 
All my sites run Ploi on Digital Ocean droplets.

I'll update the third site to `3.3.3` and see if that makes any difference.

**Comment by stephenmeehanuk** [comment]
This is in the logs `(13: Permission denied)`

```
2022/03/25 14:51:19 [crit] 32591#32591: *20560 stat() "/home/preview-k7edp/preview.website.co.uk/public/img/containers/assets/projects/south-heywood-sustainable-urban-extension/chester-wade-msnyz9L6gs4-unsplash.jpg/70f497dae9551cc902d37902d763cb0a.webp" failed (13: Permission denied), client: 86.28.25.167, server: preview.website.co.uk, request: "GET /img/containers/assets/projects/south-heywood-sustainable-urban-extension/chester-wade-msnyz9L6gs4-unsplash.jpg/70f497dae9551cc902d37902d763cb0a.webp HTTP/2.0", host: "preview.website-ldp.co.uk", referrer: "https://preview.website.co.uk/our-projects/south-heywood-sustainable-urban-extension"
```

**Reply by jasonvarga** [reply]
Looks like a permission issue then. Can you try to delete that image? and/or set the directory as writable?

**Reply by stephenmeehanuk** [reply]
Yes, I can (and just have) deleted the image.

**Reply by stephenmeehanuk** [reply]
How would I set a directory as writable? I (guess wrongly) if I can upload and delete images the directory is writeable, no?

**Reply by jasonvarga** [reply]
If you try to visit whatever page was causing that image to be generated, does it work now?

**Reply by stephenmeehanuk** [reply]
I'm using glide presets set in assets.php, so the presets are generated when the image is uploaded.

**Reply by stephenmeehanuk** [reply]
I'm going to park this here today, I'll take a look at it in the morning - with fresh eyes. Thanks for your help so far.
